plugins/modules/source_control/gitlab/gitlab_protected_branch.py

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2021, Werner Dijkerman (ikben@werner-dijkerman.nl)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+module: gitlab_protected_branch
+short_description: Creates/deletes proteched branches in Gitlab for an project.
+description:
+  - Create/Delete a protected branch in an project.
+author:
+  - "Werner Dijkerman (@dj-wasabi)"
+requirements:
+  - python >= 2.7
+  - python-gitlab python module
+extends_documentation_fragment:
+- community.general.auth_basic
+
+options:
+  state:
+    description:
+      - Create or delete proteced branch.
+      - Possible values are present and absent.
+    default: present
+    type: str
+    choices: ["present", "absent"]
+  api_token:
+    description:
+      - GitLab access token with API permissions.
+    required: true
+    type: str
+  project:
+    description:
+      - The path and name of the project.
+    required: true
+    type: str
+  name:
+    description:
+      - The name of the branch that needs to be protected.
+      - Can make use a wildcard charachter for like 'production/*' or just have 'main' or 'develop' as value.
+    required: true
+    type: str
+  merge_access_levels:
+    description:
+      - Access levels allowed to merge.
+      - Possible values are "maintainer", "developer" or "no_access".
+    default: maintainer
+    type: str
+    choices: ["maintainer", "developer", "no_access"]
```

Comment:
Wouldn't `none` or `nobody` be a better choice than `no_access`?

Reply:
I used "no_access" as there is also this one in the python library: `gitlab.NO_ACCESS`

Reply:
The GitLab API seems to be famous for using not very helpful naming :) I guess it's better to use another name here.

Reply:
Well, it's a suggestion and my personal opinion. It would be nice if some more people could chime in with their opinions.

@Lunik @Shaps @marwatk @metanovii @scodeman @waheedi @zanssa you are maintainers of GitLab modules, could you plase take a look at this question (or this PR in general)?